### DISCUSSION TITLE
using sfdisk to generate a hybrid/protective nested mbr table

### DISCUSSION THREAD

**Discussion by greyltc** [util-linux/util-linux#3283] [Q&A]
Hi, let's say I start with a disk that has a GPT with two partitions. Is there an easy/simple way for me to use sfdisk to generate and write a nested hybrid/protective mbr table that is in sync with the GPT one? Basically I'm looking for something like
```
sfdisk --label-nested mbr --sync-to-gpt /dev/sda
```
or
```
sfdisk --label gpt --dump-for-hybrid /dev/sda > for_hybrid.txt
cat for_hybrid.txt | sfdisk --label-nested mbr /dev/sda
```
other tools I'm aware of that can do a similar thing are gptsync[1] and gdisk's `h` option[2]. Maybe the only slightly weird thing going on here is gpt --> mbr partition type mapping.

Cheers!

[1]: https://manpages.ubuntu.com/manpages/trusty/man8/gptsync.8.html
[2]: https://www.rodsbooks.com/gdisk/gdisk.html

**Comment by karelzak** [comment]
You can force sfdisk to work with nested protective MBR and define MBR partitions manually. sfdisk will not do any calculation from GPT to MBR, because it's impossible in a reliable way.

**Comment by greyltc** [comment]
>  impossible in a reliable way

Yeah, an option like `--label gpt --dump-for-hybrid` would have to be "best effort"
